server/storage/backend/verify.go

```diff
@@ -67,3 +69,41 @@ func insideUnittest() bool {
 	stackTraceStr := string(debug.Stack())
 	return strings.Contains(stackTraceStr, "_test.go") && !strings.Contains(stackTraceStr, "tests/")
 }
+
+// VerifyBackendConsistency verifies data in ReadTx and BatchTx are consistent.
+func VerifyBackendConsistency(b Backend, lg *zap.Logger, skipSafeRangeBucket bool, bucket ...Bucket) {
+	verify.Verify(func() {
+		if b == nil {
+			return
+		}
+		if lg != nil {
+			lg.Debug("verifyBackendConsistency", zap.Bool("skipSafeRangeBucket", skipSafeRangeBucket))
+		}
+		b.BatchTx().LockOutsideApply()
```

Comment:
Are we sure we should use LockOutsideApply? We call this function in applyAll

Reply:
yes, LockInsideApply would fail.

Reply:
We should use `LockOutsideApply`. This is just verification code, obviously we don't need to execute the [txPostLockInsideApplyHook](https://github.com/etcd-io/etcd/blob/cf7043391c624afb5be879592ce24a136840a1bd/server/storage/backend/backend.go#L73).